### ISSUE TITLE
Linux firmware package won't upgrade

### ISSUE THREAD

**Issue by comradedakota** [AbelCS/ubuntu-server-bionic-rpi-3-b-plus#4]
For some reason, I can't seem to upgrade the Linux firmware package. After I try to run `sudo apt-get update && sudo apt-get dist-upgrade -y`, I get the following output:

```
The following packages will be upgraded:
  linux-firmware
1 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
95 not fully installed or removed.
Need to get 0 B/64.6 MB of archives.
After this operation, 30.7 kB of additional disk space will be used.
(Reading database ... 125114 files and directories currently installed.)
Preparing to unpack .../linux-firmware_1.173.2_all.deb ...
Unpacking linux-firmware (1.173.2) over (1.173.1) ...
dpkg: error processing archive /var/cache/apt/archives/linux-firmware_1.173.2_all.deb (--unpack):
 trying to overwrite '/lib/firmware/brcm/brcmfmac43455-sdio.bin', which is the diverted version of '/tmp/brcmfmac43455-sdio.bin'
dpkg-deb: error: paste subprocess was killed by signal (Broken pipe)
update-initramfs: Generating /boot/initrd.img-4.15.0-1022-raspi2
Using DTB: bcm2710-rpi-3-b-plus.dtb
Installing /lib/firmware/4.15.0-1022-raspi2/device-tree/bcm2710-rpi-3-b-plus.dtb into /boot/dtbs/4.15.0-1022-raspi2/bcm2710-rpi-3-b-plus.dtb
Taking backup of bcm2710-rpi-3-b-plus.dtb.
Installing new bcm2710-rpi-3-b-plus.dtb.
Ignoring old or unknown version 4.15.0-1022-raspi2 (latest is 4.15.0-1030-raspi2)
update-initramfs: Generating /boot/initrd.img-4.15.0-1017-raspi2
Using DTB: bcm2710-rpi-3-b-plus.dtb
Installing /lib/firmware/4.15.0-1017-raspi2/device-tree/bcm2710-rpi-3-b-plus.dtb into /boot/dtbs/4.15.0-1017-raspi2/bcm2710-rpi-3-b-plus.dtb
Taking backup of bcm2710-rpi-3-b-plus.dtb.
Installing new bcm2710-rpi-3-b-plus.dtb.
Ignoring old or unknown version 4.15.0-1017-raspi2 (latest is 4.15.0-1030-raspi2)
Errors were encountered while processing:
 /var/cache/apt/archives/linux-firmware_1.173.2_all.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

Any help would be greatly appreciated. Thanks!

**Comment by gewinh** [comment]
Exactly the same problem here. 
Everything else working brilliantly.
Would be great to get this issue fixed. Thanks

**Comment by bjmi** [comment]
You could also try 18.04 beta image for Raspberry 3
http://cdimage.ubuntu.com/releases/18.04/beta/

My Raspberry 3 B+ works with this image and was upgraded to 18.10 successfully.

**Comment by AbelCS** [comment]
Sorry, I'm away, so I have not been able to check it these days. As soon as I return I'll try to solve it.

**Comment by gewinh** [comment]
Thanks bjmi I did try the image in your link on the 3 B+ and it worked fined, installed all the updates no problem. AbelCS thanks for your reply. Interested to know what the difference is between the 2 images and which will be better for long term support.

**Comment by gewinh** [comment]
Unfortunately no perfect solution .  The image http://cdimage.ubuntu.com/releases/18.04/beta/ and I assume the ABelCS image have networking issues on the pi 3B +, on large file tranfers ie backups using rsync the system freezes. Attempts to get around this, creating a swap file, disabling segmentatiion offload didn't work. I assume there are problems with the lan78xx 1-1.1.1:1.0 driver. https://www.raspberrypi.org/forums/viewtopic.php?t=209258
Booting the same SD card/image on a pi 3 does not show any of these problems. it uses the  smsc95xx 1-1.1:1.0 driver. Backups go faster and without problems.

**Comment by AbelCS** [comment]
Hi, sorry for the delay, I've been busy lately. 

I've just published a new release for [18.04.2](https://github.com/AbelCS/ubuntu-server-bionic-rpi-3-b-plus/releases/tag/18.04.2-v1.0.0) in which I've fixed the upgrade conflicts and I think ethernet problems (#2) must be fixed too.

Can you try it and give me some feedback?


Thanks!

**Comment by gewinh** [comment]
Abel

Thanks for your working doing this. Unfortunately I haven't got a test environment now. I have moved to the official pi3 images on http://cdimage.ubuntu.com/releases/18.04/release/ and it seems to be working fine on the 3+. I'm using it to run a openvpn server and it seems very stable.

Gavin
Index of /releases/18.04/release - cdimage.ubuntu.com<http://cdimage.ubuntu.com/releases/18.04/release/>
CD images for Ubuntu 18.04.2 LTS (Bionic Beaver) Server install image. The server install image allows you to install Ubuntu permanently on a computer for use as a server.
cdimage.ubuntu.com



________________________________
